### PR TITLE
scr_param: escapes braces in regex for embedded variables

### PR DESCRIPTION
**Pull request by adammoody**
Also adds a unit test for scr_param.pm functionality

### CODE REVIEW

**Comment by adammoody** [comment]
Will need rebase on https://github.com/LLNL/scr/pull/104

File: scripts/test/test_scr_param.in [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+# test a parameter that uses embedded variables names with braces
+$name = "SCR_PATH_TEST";
+$ENV{$name} = "/\${TESTVAR}/\${TESTVAR1}/\${TESTVAR_2}/thing";
+$expected_value = "/testvar/testvar1/testvar_2/thing";
+$value = $param->get($name);
+if ($value ne $expected_value) {
+  print "scr_param->get($name)=$value intead of $expected_value\n";
+  $rc = 1;
+}
```
Comment (rhaas80) [inline scripts/test/test_scr_param.in]: I'd add a test for more unusual cases as well:
* improperly balances braces: `a/${FOO/bar`
* $ at the end of a line: `abc$`
* non-existing variable `$FAKEVAR`

**Comment by adammoody** [comment]
Thanks for the review @rhaas80 .  I agree with all of your suggestions above.  However, I won't have the time to implement them.  Mind if I had this over to you to be extended?

**Comment by rhaas80** [comment]
Sure.

**Comment by adammoody** [comment]
Great!  Thanks, @rhaas80 !

**Comment by rhaas80** [comment]
I have added the suggested checks as well as a check for getting a hash value in rhaas80/scr@32f26c9ef109611b3ecec6d45479e69db7cd9448 in branch regex in my clone.
I did:
* add the tests suggested above
* add test for getting a hash
* fixed a bug that would the code match unbalanced braces
* fixed how `get_hash` returns a hash from an env (was aborting due to incorrect assignments before)

Do you want a pull request into your branch or just pull over my changes into your branch or want me to make a new pull request from my branch?

**Comment by adammoody** [comment]
Perfect.  Go ahead and submit a new PR that contains everything.  We could close this old PR out.  Thanks again, @rhaas80 !

**Comment by rhaas80** [comment]
Updates pull request is here: https://github.com/LLNL/scr/pull/151

**Pull request closed without merging**